create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -70,6 +75,22 @@ private CreatePublicationRequest generateCreatePublicationRequest(DocTp docTp) {
                 .getItemid();
     }
 
+    private EntityDescription generateEntityDescription(DocTp docTp) throws URISyntaxException {
+        EntityDescription entityDescription = new EntityDescription();
+        entityDescription.setReference(generateReference(docTp));
+        return  entityDescription;
+    }
+
+    private Reference generateReference(DocTp docTp) throws URISyntaxException {
+        Reference reference = new Reference();
+        reference.setDoi(extractDOI(docTp));
+        return reference;
+    }
+
+    private URI extractDOI(DocTp docTp) throws URISyntaxException {
+        return new URI(DOI_OPEN_URL_FORMAT + "/" + docTp.getMeta().getDoi() );
```

Comment:
You can avoid all this if  you use UriWrapper.
```
return new UriWrapper("https","doi.org").addChild(docTp.getMeta().getDoi()).getUri() 
```
I suggest you also edit UriWrapper in commons to create a static const for "https" or better a constructor where the "https" is the default scheme. Of course all magic strings should be constants.

The reason I am suggesting UriWrapper is because we probably have around 100 static constants for "/" with various names such as SLASH, DELIMITER, SEPARATOR, PATH_DELIMITER etc,

Reply:
Fiksed